modules/rounds.js

```diff
@@ -170,7 +170,21 @@ Rounds.prototype.backwardTick = function(block, previousBlock, done, tx) {
 		err => {
 			// Stop round ticking
 			__private.ticking = false;
-			return done(err);
+			if (err) {
+				return done(err);
+			}
```

Comment:
@lsilvs I am not so sure if the block has been deleted because one of the previous steps could fail as well. For me it's confusing because `round.backwardLand()` consists of many chained methods and it's not possible to determine which one failed. So like you addressed keeping cache can cause a problem. 

So maybe instead of clearing cache in every error i can put another call before line 166 and clear the cache only in that catch block? What do you think?

Reply:
I will leave this one to @MaciejBaj and @4miners as it's not clear for me as well.
But I think we should have tests covering all these scenarios in this PR so that we can ensure it's working as expected.

Reply:
Yep, a block is already deleted ("popped"). If backwardTick fails, it would mean that a previous round was unsuccessfully restored, and because the steps were in the same database transaction, the block will be restored back. Cached delegates may be reused, so it's fine like this.